### PR TITLE
[Fix] gmx-gpl race condition (continue)

### PR DESCRIPTION
**Pull request by phqb**

### CODE REVIEW

File: pkg/source/gmx-glp/pool_simulator_stake.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ func (p *PoolSimulator) addLiquidityForAccount(swapInfo *gmxGlpSwapInfo, tokenIn
 
 func (p *PoolSimulator) BuyUSDG(swapInfo *gmxGlpSwapInfo, token string, tokenAmount *big.Int) (*big.Int, error) {
 	//_validate(whitelistedTokens[_token], 16);  // canSwapTo vaildated it
-	p.vault.UseSwapPricing = true
```
Comment (it4rb) [inline pkg/source/gmx-glp/pool_simulator_stake.go:58]: please remove this field in vault struct as well

File: pkg/source/gmx-glp/yearn_token_vault.go [hunk 2]
```diff
@@ -52,22 +69,30 @@ func (y *YearnTokenVault) Deposit(amount *big.Int) (*big.Int, error) {
...
-func (y *YearnTokenVault) Withdraw(maxShares *big.Int) (*big.Int, error) {
+func (y *YearnTokenVault) Withdraw(maxShares *big.Int) (*big.Int, *YearnTokenVault, error) {
 	shares := new(big.Int).Set(maxShares)
 	if shares.Cmp(bignumber.ZeroBI) <= 0 {
-		return nil, ErrYearnTokenVaultWithdrawNothing
+		return nil, nil, ErrYearnTokenVaultWithdrawNothing
 	}
 
 	value := y.shareValue(shares)
 	vaultBalance := new(big.Int).Set(y.TotalIdle)
```
Comment (it4rb) [inline pkg/source/gmx-glp/yearn_token_vault.go:79]: will this function get call multiple times within 1 swap? if so then maybe we'll need to check and use `yModified.TotalIdle` here

Comment (phqb) [reply pkg/source/gmx-glp/yearn_token_vault.go:79]: Withdraw() is called once at CalcAmountOut(). But I lifted `yModified` out of `Withdraw()` to make `Withdraw()` able to be called multiple times.

File: pkg/source/gmx-glp/pool_simulator_stake.go (hunk 1 above)
Comment (phqb) [reply pkg/source/gmx-glp/pool_simulator_stake.go:58]: done

**Review by it4rb — APPROVED** [review]

**Pull request merged**